Automattic/newspack-theme
# feat: add optional X icon

### All Submissions:


### Changes proposed in this Pull Request:

Following how this is being handled in Jetpack's Share buttons and Core, this PR adds an option to use the X icon in the social links menu by using an `x.com` domain to link to your profile, while keeping the original Twitter icon for profiles using `twitter.com`.

See: 200550061930446-as-1205128714156146

### How to test the changes in this Pull Request:

1. Apply the PR.
2. Edit your social links menu to make sure it's showing; add at least one twitter.com user profile link, and one x.com user profile link (they can both be for the same account, using different URLs - eg `https://twitter.com/automattic` and `https://x.com/automattic`.)
3. Confirm that you're getting the right icon for each -- the bird for the twitter.com link, and the X for the x.com link:

![image](https://github.com/Automattic/newspack-theme/assets/177561/919976fa-2e01-4dbd-a8f6-00c775ea0c83)

4. Test against a dark background if you're currently testing against white, or vice versa, to make sure the icon picks up the right colours. 

### Other information:
